tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/DataConnections.resource

```diff
@@ -0,0 +1,74 @@
+    Click Button    ${DC_ADD_BTN_1_XP} 
+    Wait Until Page Contains Element    ${S3_NAME_DC_INPUT_XP}
+    Input Text    ${S3_NAME_DC_INPUT_XP}    ${dc_name}
+    Input Text    ${S3_KEY_DC_INPUT_XP}    ${aws_access_key}
+    Input Text    ${S3_SECRET_DC_INPUT_XP}    ${aws_secret_access}
+    Input Text    ${S3_ENDPOINT_DC_INPUT_XP}    ${aws_s3_endpoint}
+    Input Text    ${S3_REGION_DC_INPUT_XP}    ${aws_region}
+    IF    ${press_cancel} == ${TRUE}
+        Click Button    ${GENERIC_CANCEL_BTN_XP}
+    ELSE
+        Wait Until Element Is Enabled    ${STORAGE_ADD_BTN_2_XP}
+        Click Button    ${STORAGE_ADD_BTN_2_XP}
+    END
+    Wait Until Modal Disappears
+    Wait Until Project Is Open    project_title=${project_title}
+
+Data Connection Should Be Listed
+    [Documentation]    Checks a Data Connection is listed in DS Project details page
+    [Arguments]     ${name}   ${type}   ${connected_workbench}
+    Run keyword And Continue On Failure     Page Should Contain Element    
+    ...    ${DC_SECTION_XP}//td[@data-label="Name"]/h4/div[text()="${name}"]
+    Run keyword And Continue On Failure     Page Should Contain Element
+    ...    ${DC_SECTION_XP}//tr[td[@data-label="Name"]/h4/div[text()="${name}"]]/td[text()=" ${type}"]
+    IF    "${connected_workbench}" == "${NONE}"
+        Run Keyword And Continue On Failure    Page Should Contain Element    
+        ...    ${DC_SECTION_XP}//tr[td[@data-label="Name"]/h4/div[text()="${name}"]]/td[text()="No connections"]
+    ELSE
+        FOR    ${index}    ${workbench_title}    IN ENUMERATE    @{connected_workbench}
+            Log    ${index}: ${workbench_title}
+            Run Keyword And Continue On Failure    Page Should Contain Element    
+            ...    ${DC_SECTION_XP}//tr[td[@data-label="Name"]/h4/div[text()="${name}"]]/td[@data-label="Connected workbenches"]/ul/li[text()="${workbench_title}"]
+        END
+    END
+
+Data Connection Should Not Be Listed
+    [Documentation]    Checks a Data Connection is not listed in DS Project details page
+    [Arguments]     ${name}
+    Run keyword And Continue On Failure     Wait Until Page Does Not Contain Element    
+    ...    ${DC_SECTION_XP}//tr[td[@data-label="Name"]/h4/div[text()="${name}"]]
+
+Get Openshift Secret From Data Connection
+    [Documentation]    Retrieves name of Openshift secret corresponding to a given S3 Data Connection based on Dispayed name in DS Project details page
+    [Arguments]     ${dc_name}      ${namespace}
+    ${rc}  ${secret_name}=    Run And Return Rc And Output   oc get secret -n ${namespace} -o jsonpath='{.items[?(@.metadata.annotations.openshift\\.io/display-name=="${dc_name}")].metadata.name}'
+    [Return]    ${rc}    ${secret_name}
+
+Delete Data Connection
+    [Documentation]    Deletes a Data Connection from DS Project details page
+    [Arguments]     ${name}    ${press_cancel}=${FALSE}
```

Comment:
about the ${dc_name}, pls read the comment above https://github.com/red-hat-data-services/ods-ci/pull/583#discussion_r1029613995 

${press_cancel} is needed only for specific tests, so I thought these keywords will be re-used for other scenarios (e.g., model serving), then it would spare an argument (very likely always set to FALSE) to the developer who is gonna them